src/metabase/db/migrations.clj

```diff
@@ -331,7 +331,11 @@
 ;; missing those database ids
 (defmigration ^{:author "senior", :added "0.27.0"} populate-card-database-id
   (doseq [[db-id cards] (group-by #(get-in % [:dataset_query :database])
-                                  (db/select [Card :dataset_query :id] :database_id [:= nil]))
+                                  (db/select [Card :dataset_query :id :name] :database_id [:= nil]))
           :when (not= db-id virtual-id)]
-    (db/update-where! Card {:id [:in (map :id cards)]}
-      :database_id db-id)))
+    (if (db/exists? Database :id db-id)
+      (db/update-where! Card {:id [:in (map :id cards)]}
```

Comment:
You should check `(seq cards)` before doing this because otherwise HoneySQL will generate SQL like `WHERE id IN (NULL)` which is invalid and will barf